### PR TITLE
Refactoring in ContainerRunner

### PR DESCRIPTION
**Pull request by nulls**
### What's done:
- combined cleanup[ByExecution] and stop[ByExecution]
- added ContainerRunner.Prunable and ContainerRunner.Stoppable -- only docker does implement it
- removed discover method

### CODE REVIEW

File: save-orchestrator-common/src/main/kotlin/com/saveourtool/save/orchestrator/docker/DockerContainerRunner.kt [hunk 1]
```diff
@@ -72,7 +72,7 @@ class DockerContainerRunner(
         }
     }
 
-    override fun start(executionId: Long) {
+    override fun startAllByExecution(executionId: Long) {
```
Comment (nulls) [inline save-orchestrator-common/src/main/kotlin/com/saveourtool/save/orchestrator/docker/DockerContainerRunner.kt:75]: will be removed in next PR

File: save-orchestrator-common/src/main/kotlin/com/saveourtool/save/orchestrator/kubernetes/KubernetesManager.kt [hunk 2]
```diff
@@ -98,51 +98,25 @@ class KubernetesManager(
             .orEmpty()
     }
 
-    override fun start(executionId: Long) {
+    override fun startAllByExecution(executionId: Long) {
```
Comment (nulls) [inline save-orchestrator-common/src/main/kotlin/com/saveourtool/save/orchestrator/kubernetes/KubernetesManager.kt:101]: will be removed in next PR

File: save-orchestrator-common/src/main/kotlin/com/saveourtool/save/orchestrator/service/ContainerService.kt [hunk 3]
```diff
@@ -97,7 +98,8 @@ class ContainerService(
                     .doOnComplete {
                         if (areAgentsHaveStarted[executionId]?.get() != true) {
                             log.error("Internal error: none of agents $containerIds are started, will mark execution $executionId as failed.")
-                            containerRunner.stop(executionId)
+                            cleanup(executionId)
```
Comment (nulls) [inline save-orchestrator-common/src/main/kotlin/com/saveourtool/save/orchestrator/service/ContainerService.kt:101]: duplicate calls, need to remove one of them

Comment (nulls) [reply save-orchestrator-common/src/main/kotlin/com/saveourtool/save/orchestrator/service/ContainerService.kt]: fixed

File: save-orchestrator-common/src/main/kotlin/com/saveourtool/save/orchestrator/docker/DockerContainerRunner.kt [hunk 4]
```diff
@@ -113,13 +103,13 @@ class DockerContainerRunner(
...
 
-    override fun isStoppedByContainerId(containerId: String): Boolean = dockerClient.inspectContainerCmd(containerId)
+    override fun isStopped(containerId: String): Boolean = dockerClient.inspectContainerCmd(containerId)
         .exec()
         .state
         .also { logger.debug("Container $containerId has state $it") }
         .status != "running"
 
-    override fun cleanup(executionId: Long) {
+    override fun cleanupAllByExecution(executionId: Long) {
```
[the remark this answers is not in the document]
Comment (nulls) [reply save-orchestrator-common/src/main/kotlin/com/saveourtool/save/orchestrator/docker/DockerContainerRunner.kt:112]: `logs` are stored in `loki` and `agent` doesn't have any state -- nothing should be lost

[the remark this answers is not in the document]
Comment (nulls) [reply save-orchestrator-common/src/main/kotlin/com/saveourtool/save/orchestrator/docker/DockerContainerRunner.kt:112]: also, it's called only in `ContainerService#startContainersAndUpdateExecution` when no agents are started for execution

**Review by kgevorkyan — APPROVED** [review]

**Pull request merged**